src/main/scala/org/apache/spark/sql/execution/datasources/greenplum/GreenplumOptions.scala

```diff
@@ -45,6 +46,10 @@ case class GreenplumOptions(
   /** Max number of times we are allowed to retry dropTempTable operation. */
   val dropTempTableMaxRetries: Int = 3
 
+  /** Timeout for copying a partition's data to greenplum. */
+  val copyTimeout = Utils.timeStringAsMs(params.getOrElse("copyTimeout", "1h"))
+  assert(copyTimeout > 0, "The copy timeout should be a positive number.")
```

Comment:
The copyTimeout should be positive, 10s, 10min, 1h etc.

Reply:
done.